### ISSUE TITLE
Ansible will not find tasks in a playbook with hosts field set to 127.0.0.1

### ISSUE THREAD

**Issue by willowmck** [ansible/ansible#17288]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

TaskQueueManager
##### ANSIBLE VERSION

```
2.1.1.0
```
##### OS / ENVIRONMENT

Mac OS/X 10.11.5
##### SUMMARY

We have a simple one task playbook that waits for port availability on localhost.  If we set the hosts field to 127.0.0.1, the TaskQueueManager will not find any tasks.  Changing this to localhost makes it work.  This used to work in Ansible 1.9.x with 127.0.0.1 and the Ansible documentation specifies using the IP address.
##### STEPS TO REPRODUCE

Create a playbook that will wait for port availability on 127.0.0.1 and execute the playbook with Ansible debug turned on.  You will see num_tasks = 0.

```

---
- hosts: 127.0.0.1
  connection: local
  gather_facts: False

  tasks:
  - name: Check that target host & port are accessible
    wait_for: host={{ target_host }} port={{ target_port }} timeout={{ timeout }}
```

##### EXPECTED RESULTS

Expected there to be task execution.
##### ACTUAL RESULTS

No tasks found.

```

```

**Comment by robinro** [comment]
Please list your inventory here.

If you don't set the inventory yourself you rely on the default inventory, which changed between 1.9 and 2.x. Run `ansible-playbook -i 127.0.0.1, your_playbook.yml` to explicitly have the ip in the inventory (works also with 1.9).

This is not a bug, but a change in behavior to make other things (e.g. ansible-pull) more consistent. In general it's a good idea to explicitly specify your inventory.

**Comment by akamensky** [comment]
You can use `- hosts: localhost`.

It will produce warnings, but will work.

**Comment by willowmck** [comment]
Yes, we are using localhost now which works.  But, the documentation took us down the wrong path.

**Comment by acozine** [comment]
@willowmck I'm working to address old documentation tickets and ran across this one - unfortunately the original description didn't include a link to the page in the docs suggesting that you should use the IP address in the hosts field. Do you remember where that was? I'm hoping the docs were updated and by mistake this ticket wasn't closed - at any rate I haven't found a page with this information on it yet. Chime in if you know where it is/was.

**Comment by willowmck** [comment]
@acozine Apologies, I wish I had submitted my report with a link.  I can no longer find that documentation and it appears difficult to find anything 2.x prior to 2.3 anyway.

**Comment by bcoca** [comment]
This has been fixed already, I've tested 2.4/2.5 and devel and it picks up the tasks normally.

As such I'm going to close the ticket, feel free to reopen if you find further issues.
